src/compiler/sys.ts

```diff
@@ -439,12 +449,26 @@ namespace ts {
                 return buffer.toString("utf8");
             }
 
+            const outputFingerprintMap: OutputFingerprintMap = {};
+
             function writeFile(fileName: string, data: string, writeByteOrderMark?: boolean): void {
                 // If a BOM is required, emit one
                 if (writeByteOrderMark) {
                     data = "\uFEFF" + data;
                 }
 
+                const md5 = getMd5(data);

```

Comment:
i do not think we should do this for all files. in a normal patch compilation secnario, you will doing a fileExits, stat, and createHash needlessly. 

Instead, we should expose a new optional function on sys to getModifiedTime, and createHash. in the watch logic, we can query them as needed, and store the map outputFileName => { mtime,  hash} there as well.